src/main/scala/fi/vm/sade/hakurekisteri/web/kkhakija/KkExcelUtilV5.scala

```diff
@@ -0,0 +1,236 @@
+    "Hyväksymisen ehto (ehdolliisestiHyvaksyttavissa,ehtoKoodi,ehtoFI,ehtoSV,ehtoEN)",
+    "Valintatapajonon tyyppi",
+    "Valintatapajonon nimi",
+    "Vastaanottotieto",
+    "Ilmoittautumiset",
+    "Pohjakoulutus",
+    "Julkaisulupa",
+    "Hakukelpoisuus",
+    "Hakukelpoisuuden lahde",
+    "Maksuvelvollisuus",
+    "Hakukohteen koulutukset 1(komoOid,koulutusKoodi,kkKoulutusId,koulutuksenAlkamisvuosi,koulutuksenAlkamiskausi," +
+      "johtaaTutkintoon)",
+    "Koulutus 2",
+    "Koulutus 3",
+    "Koulutus 4",
+    "Koulutus 5",
+    "Koulutus 6",
+    "Liite 1(hakuId,hakuRyhmäId,tila,saapumisenTila,nimi,vastaanottaja)",
+    "Liite 2",
+    "Liite 3",
+    "Liite 4",
+    "Liite 5",
+    "Liite 6"
+  )
+
+  override def getHeaders(hakijat: Seq[Hakija]): Set[Row] = Set(
+    Row(0, headers.zipWithIndex.toSet.map((h: (String, Int)) => StringCell(h._2, h._1)))
+  )
+
+  override def getRows(hakijat: Seq[Hakija]): Set[Row] = hakijat
+    .flatMap(hakija =>
+      hakija.hakemukset.map(hakemus => {
+        val rivi = Seq(
+          hakija.hetu,
+          hakija.syntymaaika.getOrElse(""),
+          hakija.oppijanumero,
+          hakija.sukunimi,
+          hakija.etunimet,
+          hakija.kutsumanimi,
+          hakija.lahiosoite,
+          hakija.postinumero,
+          hakija.postitoimipaikka,
+          hakija.maa,
+          hakija.kansalaisuudet.getOrElse(List.empty).mkString(", "),
+          hakija.matkapuhelin.getOrElse(""),
+          hakija.puhelin.getOrElse(""),
+          hakija.sahkoposti.getOrElse(""),
+          hakemus.lukuvuosimaksu.getOrElse(""),
+          hakija.kotikunta,
+          hakija.sukupuoli,
+          hakija.aidinkieli,
+          hakija.asiointikieli,
+          hakija.koulusivistyskielet.getOrElse(List.empty).mkString(", "),
+          toBooleanX(hakija.koulutusmarkkinointilupa),
+          toBooleanX(hakija.onYlioppilas),
+          hakija.yoSuoritusVuosi.getOrElse(""),
+          hakija.ensikertalainen.map(e => toBooleanX(e)).getOrElse(""),
+          hakemus.haku,
+          hakemus.hakuVuosi.toString,
+          hakemus.hakuKausi,
+          hakemus.hakemusnumero,
+          hakemus.hakemusJattoAikaleima.getOrElse(""),
+          hakemus.hakemusViimeinenMuokkausAikaleima.getOrElse(""),
+          hakemus.organisaatio,
+          hakemus.hakukohde,
+          hakemus.hakukohdeKkId.getOrElse(""),
+          hakemus.hakutoivePrioriteetti.map(_.toString).getOrElse(""),
+          toBooleanX(hakemus.avoinVayla),
+          hakemus.valinnanTila.map(_.toString).getOrElse(""),
+          hakemus.valinnanAikaleima.getOrElse(""),
+          hakemus.pisteet.map(_.toString()).getOrElse(""),
+          hakemus.hyvaksymisenEhto match {
+            case None => ""
+            case Some(h) =>
+              s"HyvaksymisenEhto(${toBooleanX(h.ehdollisestiHyvaksyttavissa)},${h.ehtoKoodi.getOrElse("")}," +
+                s"${h.ehtoFI.getOrElse("")},${h.ehtoSV.getOrElse("")},${h.ehtoEN.getOrElse("")})"
```

Comment:
Haluttiinkohan tässä kaikki käännökset mukaan, vai riittäisiköhän yksi löytyvä jossain prioriteettijärjestyksessä? Voi tietty olla, että on hyvä näin. Mut tässähän voi käydä niin, että tulee tyhjiä pilkkuvälejä jos optiot on tyhjiä, mikä voi olla vähän rumaa.

Reply:
Noita hyväksymisen ehtoja on onneksi vain muutama ja kaikille on kaikki käännökset määriteltynä: https://virkailija.testiopintopolku.fi/koodisto-app/koodisto/view/hyvaksynnanehdot/1

Luulen siis että siisti näin, tietysti herää kysymys mihin noita käännöksiä tarvitaan kun koodistokin on niin pieni.

... siis: jos tämän mallintaisi oikein siististi, niin kaiketi hyväksymisen ehto kyllä / ei -tieto olisi pakollinen, tuo ehtokoodi + käännökset -kokonaisuus option (silloin kun hyväksymisen ehto on), ja sen sisällä kaikki tiedot pakollisia, tms.